magento/magento2
# Added a check quote is_active or not, because it removed quotes

### Description (*)

Somising went wrong while saving the customer.

- Magento version 2.4.3
- Anything else that would help a developer reproduce the bug:
  1. You need to have an order that was completed more than the number of days specified in the admin panel in the field Quote Lifetime (days).
  2. This order must also not be edited for the same number of days.
  3. You must have cron job enabled.

1. Go to the user who placed an order more than 30 days ago.
2. Try changing any of the fields.


The new data is not saved and I get the following error:
![CleanShot 2022-11-18 at 09 40 20](https://user-images.githubusercontent.com/125371128/218820460-e98d5d41-664b-4595-aeb8-4f07b91d26d0.jpeg)


After starting to look for the cause of this problem, I went to look in the var/log/system.log file where I found it:
![Снимок экрана от 2023-02-14 19-43-00](https://user-images.githubusercontent.com/125371128/218820891-8a6c9371-5c71-4bda-ad30-fbbf0d061fea.png)
Next I went to the database to find quote with entity_id = 173008 in quote table but it was not there.
Then I started looking for the reason why this record disappeared. 
I found the problem in vendor/magento/module-sales/Model/ResourceModel/Collection/ExpiredQuotesCollection.php
Function of this class getExpiredQuotes receives the collection of quotes and checks whether the quota is older than the number of days specified in one of the fields in the quote lifetime field but dont have check for quote is_active:
![Снимок экрана от 2023-02-14 20-12-47](https://user-images.githubusercontent.com/125371128/218823123-487f3d38-dc3a-4e39-9d3e-64525bd32317.png)
It turns out that the function returns all quotas that are older than n-number of days and then deletes them in the class -vendor/magento/module-sales/Cron/CleanExpiredQuotes.php

It turns out that without a check for quote is_active 1, Magenta uses cron to delete all fields of the quote table that are older than n-number of days, so when user data is changed, Magenta cannot find the quota associated with it and issues an error.
The solution to this problem is to add a check to see if the quota is active:
![Снимок экрана от 2023-02-14 20-26-08](https://user-images.githubusercontent.com/125371128/218825083-d170c04f-d95b-4f3f-b225-30a0ee746006.png)


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

To test these changes you need:
1. Go to database - quote table
2. Update the `created_at` and `update_at` field values ​​a couple of days ago from today's date
3. Go to admin panel Stores->Configuration->Sales->Checkout->Shopping Cart->Quote LoifeTime(days) field and set 1 day
4. Start the cron job and you will see that records that have become orders will not be deleted, which will solve the problem. Only those that have not become orders are deleted

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#36898: Added a check quote is_active or not, because it removed quotes
